deploy/buildtools/buildafi.py

```diff
@@ -67,51 +97,112 @@ def aws_build(global_build_config, bypass=False):
+    fat_jar = get_sim_makevar(buildconfig, 'FAT_JAR')
 
     # first, copy aws-fpga to the build instance. it will live in
-    # firesim-build/platforms/f1/
+    # the same path on the build host as on the manager
     with StreamLogger('stdout'), StreamLogger('stderr'):
-        run('mkdir -p /home/centos/firesim-build/platforms/f1/')
+        run('mkdir -p ' + ddir)
+        run('mkdir -p ' + generated_dir)
+        run('mkdir -p ' + dirname(fat_jar))
+        run('mkdir -p ' + aws_fpga_root)
+        run('mkdir -p ' + pjoin(aws_fpga_root, fpgatemplatedir))
+        run('mkdir -p ' + pjoin(aws_fpga_root, fpgabuilddir))
+
     # do the rsync, but ignore any checkpoints that might exist on this machine
     # (in case builds were run locally)
     # extra_opts -l preserves symlinks
     with StreamLogger('stdout'), StreamLogger('stderr'):
-        rsync_cap = rsync_project(local_dir=ddir + "/../platforms/f1/aws-fpga",
-                      remote_dir='/home/centos/firesim-build/platforms/f1/',
+        rsync_cap = rsync_project(local_dir=pjoin(aws_fpga_root,''),
+                      remote_dir=aws_fpga_root,
                       ssh_opts="-o StrictHostKeyChecking=no",
                       exclude="hdk/cl/developer_designs/cl_*",
                       extra_opts="-l", capture=True)
         rootLogger.debug(rsync_cap)
         rootLogger.debug(rsync_cap.stderr)
-        rsync_cap = rsync_project(local_dir=ddir + "/../platforms/f1/aws-fpga/{}/*".format(fpgabuilddir),
-                      remote_dir='/home/centos/firesim-build/platforms/f1/aws-fpga/' + remotefpgabuilddir,
-                      exclude='build/checkpoints',
-                      ssh_opts="-o StrictHostKeyChecking=no",
-                      extra_opts="-l", capture=True)
-        rootLogger.debug(rsync_cap)
-        rootLogger.debug(rsync_cap.stderr)
+
+    if global_build_config.distributed_elaboration:
+	with StreamLogger('stdout'), StreamLogger('stderr'):
+	    rsync_cap = rsync_project(local_dir=pjoin(aws_fpga_root, fpgatemplatedir, ''),
+		  remote_dir=pjoin(aws_fpga_root, fpgatemplatedir),
+		  exclude='build/checkpoints',
+		  ssh_opts="-o StrictHostKeyChecking=no",
+		  extra_opts="-l", capture=True)
+	    rootLogger.debug(rsync_cap)
+	    rootLogger.debug(rsync_cap.stderr)
+	    # sync the generated_dir (in a clean build, it is only the replace-rtl.sh)
+	    rsync_cap = rsync_project(local_dir=pjoin(generated_dir, ''),
+			  remote_dir=generated_dir,
+			  ssh_opts="-o StrictHostKeyChecking=no",
+			  extra_opts="-l", capture=True)
+	    rootLogger.debug(rsync_cap)
+	    rootLogger.debug(rsync_cap.stderr)
+	    # sync the fat jar
+	    rsync_cap = rsync_project(local_dir=fat_jar,
+			  remote_dir=fat_jar,
+			  ssh_opts="-o StrictHostKeyChecking=no",
+			  extra_opts="-l", capture=True)
+	    rootLogger.debug(rsync_cap)
+	    rootLogger.debug(rsync_cap.stderr)
+        with InfoStreamLogger('stdout'), InfoStreamLogger('stderr'):
```

Comment:
ugh.  This indenting is wrong.
```suggestion
      with InfoStreamLogger('stdout'), InfoStreamLogger('stderr'):
```